TestDefaultHarvesterGroup/assert_a_harvester_can_be_stopped_and_removed_from_bookkeeper test failure

## :broken_heart: Build Failed


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.8/detail/8.8/53//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.8/detail/8.8/53//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.8/detail/8.8/53//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.8/detail/8.8/53//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_null.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/beats-ci/transactions/view?rangeFrom=2023-05-23T16:27:45.721Z&rangeTo=2023-05-23T16:47:45.721Z&transactionName=BUILD+Beats%2Fbeats%2F8.8&transactionType=job&latencyAggregationType=avg&traceId=ddd6e19d974ea2b9d0004e490fd15059&transactionId=5c3a765622df7ef0)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2023-05-23T16:37:45.721+0000


* Duration: 127 min 46 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 58665  |
| Skipped      | 5788 |
| Total        | 64453   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->














### Steps errors [![3](https://img.shields.io/badge/3%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2F8.8/detail/8.8/53//pipeline)
  <details><summary>Expand to view the steps failures</summary>
  <p>


  
  
  
  
##### `filebeat-windows-8-windows-8 - mage build unitTest`
<ul>
<li>Took 2 min 10 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/8.8/runs/53/steps/30390/log/?start=0">here</a></li>
<li>Description: <code>mage build unitTest</code></l1>
</ul>
  
  
  
  
##### `Checks if running on a Unix-like node`
<ul>
<li>Took 0 min 0 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/8.8/runs/53/steps/26186/log/?start=0">here</a></li>

</ul>
  
  
  
  
##### `Checks if running on a Unix-like node`
<ul>
<li>Took 0 min 0 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/8.8/runs/53/steps/26189/log/?start=0">here</a></li>

</ul>
  
  </p>
  </details>

```
[2023-05-23T18:22:56.892Z] === Failed
[2023-05-23T18:22:56.892Z] === FAIL: filebeat/input/filestream/internal/input-logfile TestDefaultHarvesterGroup/assert_a_harvester_can_be_stopped_and_removed_from_bookkeeper (0.00s)
[2023-05-23T18:22:56.892Z]     harvester_test.go:104: 
[2023-05-23T18:22:56.892Z]         	Error Trace:	C:\Users\jenkins\workspace\8.8-53-fdddf600-93c7-4b3d-9cce-39d246dbac30\src\github.com\elastic\beats\filebeat\input\filestream\internal\input-logfile\harvester_test.go:104
[2023-05-23T18:22:56.893Z]         	            				C:\Users\jenkins\workspace\8.8-53-fdddf600-93c7-4b3d-9cce-39d246dbac30\src\github.com\elastic\beats\filebeat\input\filestream\internal\input-logfile\harvester_test.go:232
[2023-05-23T18:22:56.893Z]         	Error:      	Should be false
[2023-05-23T18:22:56.893Z]         	Test:       	TestDefaultHarvesterGroup/assert_a_harvester_can_be_stopped_and_removed_from_bookkeeper
[2023-05-23T18:22:56.893Z]     --- FAIL: TestDefaultHarvesterGroup/assert_a_harvester_can_be_stopped_and_removed_from_bookkeeper (0.00s)
```

--- a/filebeat/input/filestream/internal/input-logfile/harvester_test.go
+++ b/filebeat/input/filestream/internal/input-logfile/harvester_test.go
@@ -94,15 +94,13 @@ func TestReaderGroup(t *testing.T) {
 func TestDefaultHarvesterGroup(t *testing.T) {
 	source := &testSource{name: "/path/to/test"}
 
-	requireSourceAddedToBookkeeper :=
-		func(t *testing.T, hg *defaultHarvesterGroup, s Source) {
-			require.True(t, hg.readers.hasID(hg.identifier.ID(s)))
-		}
+	requireSourceAddedToBookkeeper := func(t *testing.T, hg *defaultHarvesterGroup, s Source) {
+		require.True(t, hg.readers.hasID(hg.identifier.ID(s)))
+	}
 
-	requireSourceRemovedFromBookkeeper :=
-		func(t *testing.T, hg *defaultHarvesterGroup, s Source) {
-			require.False(t, hg.readers.hasID(hg.identifier.ID(s)))
-		}
+	requireSourceRemovedFromBookkeeper := func(t *testing.T, hg *defaultHarvesterGroup, s Source) {
+		require.False(t, hg.readers.hasID(hg.identifier.ID(s)))
+	}
 
 	t.Run("assert a harvester is started in a goroutine", func(t *testing.T) {
 		var wg sync.WaitGroup
@@ -153,7 +151,8 @@ func TestDefaultHarvesterGroup(t *testing.T) {
 
 		mockHarvester := &mockHarvester{
 			onRun: harvesterRun,
-			wg:    &wg}
+			wg:    &wg,
+		}
 		hg := testDefaultHarvesterGroup(t, mockHarvester)
 		hg.tg = task.NewGroup(1, time.Second, &logp.Logger{}, "")
 
@@ -222,13 +221,16 @@ func TestDefaultHarvesterGroup(t *testing.T) {
 
 		goroutinesChecker.WaitUntilIncreased(1)
 		// wait until harvester is started
-		if mockHarvester.getRunCount() == 1 {
-			requireSourceAddedToBookkeeper(t, hg, source)
-			// after started, stop it
-			hg.Stop(source)
-			goroutinesChecker.WaitUntilOriginalCount()
-		}
-
+		require.Eventually(t,
+			func() bool { return mockHarvester.getRunCount() == 1 },
+			5*time.Second,
+			10*time.Millisecond,
+			"run count must equal one")
+		requireSourceAddedToBookkeeper(t, hg, source)
+		// after started, stop it
+		hg.Stop(source)
+		_, err := goroutinesChecker.WaitUntilOriginalCount()
+		require.NoError(t, err)
 		requireSourceRemovedFromBookkeeper(t, hg, source)
 	})
 
@@ -458,6 +460,7 @@ func (tl *testLogger) Errorf(format string, args ...interface{}) {
 	sb.WriteString(fmt.Sprintf(format, args...))
 	sb.WriteString("\n")
 }
+
 func (tl *testLogger) String() string {
 	return (*strings.Builder)(tl).String()
 }

